jme3-core/src/main/java/com/jme3/scene/debug/Grid.java

```diff
@@ -45,6 +45,10 @@
  */
 public class Grid extends Mesh {
 
+    public Grid() {
+        this(5, 5, 0.5f);
```

Comment:
If it's for serialization why not an empty constructor? the mesh data will be populated in the read() method so basically here you do the work twice and the first time with arbitrary data.

Reply:
Also, I can't remember for sure but it's possible that I modified SpiderMonkey's serialization to work with private/protected constructors.  In which case we could make this private and leave it empty.

Reply:
...and even if not, I think it's only a one or two line change to fix SM to work with private/protected constructors.  We already do a similar thing for private fields anyway.

Reply:
I think it can be used more then only for serialisation; now you can instantiate the mesh and it will have some default behaviour (2 by 2 world units in size). Just like the com.jme3.scene.debug.Wirebox for example. But I do get both your points. I'll have a peek at SM and see for a private constructor, or just an empty one.

Reply:
Any news on this one?